src/system-component-workspace/system-component.service.ts

```diff
@@ -88,6 +91,35 @@ export class SystemComponentWorkspaceService {
       trx,
     );
 
+    // Check if payload component is present
+    // if not valid throw an error regarding component not presetn
+    // else proceed with updating the Component 
+    if (systemComponent.component.componentId !== payload.componentId) {
+      throw new EaseyException(
+        new Error('Component was not found'),
+        HttpStatus.NOT_FOUND,
+        {
+          componentId: payload.componentId,
+        },
+      );
+    }
+    else
+    if (systemComponent.component.componentId!= null) {
+      // Get Component 
+      let component = await this.componentService.getComponentByIdentifier(
+        locationId,
+        payload.componentId,
+        trx,
+      );
+      component.updateDate =  new Date().toISOString();
+
+      // apply changes to the Component 
+      Object.assign(component, payload);
+
+      // update the Component entity
+      await withTransaction(this.componentWorkspaceRepository, trx).save(component);
+  }
+
```

Comment:
This works, but it works by way of skirting around TypeScript and I don't it should be done here. This works because the frontend is passing extra arguments in the payload to this endpoint that aren't being caught by TypeScript. The `ComponentWorkspaceService`, however, already has its own `updateComponent` endpoint that should be used instead.

This will require updating the `ComponentWorkspaceController` to something like the following, and then updating the UI to call both endpoints:

```typescript
export class ComponentWorkspaceController {
  constructor(
    private readonly repository: ComponentWorkspaceRepository,
    private readonly service: ComponentWorkspaceService,
    private readonly checkService: ComponentCheckService,
  ) {}

  @Put(':compId')
  @RoleGuard(
    {
      pathParam: 'locId',
      requiredRoles: ['Preparer', 'Submitter', 'Sponsor', 'Initial Authorizer'],
      permissionsForFacility: ['DSMP'],
    },
    LookupType.Location,
  )
  @ApiOkResponse({
    type: ComponentDTO,
    description: 'Updates a component record',
  })
  async updateCompnent(
    @Param('locId') locationId: string,
    @Param('compId') compId: string,
    @Body() payload: UpdateComponentBaseDTO,
    @User() user: CurrentUser,
  ): Promise<ComponentDTO> {
    return this.service.updateComponent({
      componentRecord: await this.repository.getComponentByLocIdAndCompId(
        locationId,
        compId,
      ),
      locationId,
      payload,
      userId: user.userId,
    });
  }
```

This will certainly be less succinct, but I think it's necessary to retain the type correctness and purity of the separate functions (although we could, alternatively, create an accurately-named endpoint that does both).